kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -370,17 +370,44 @@ impl<'tcx> GotocCtx<'tcx> {
             }};
         }
 
-        if let Some(_stripped) = intrinsic.strip_prefix("simd_shuffle") {
-            // TODO: can be empty now (i.e. `simd_shuffle` instead of `simd_shuffle8`)
-            // `parse` fails on empty, so comment that bit of code out.
-            // To re-enable this we'll need to investigate how size is computed now.
-            // let n: u64 = stripped.parse().unwrap();
-            return unstable_codegen!(self.codegen_intrinsic_simd_shuffle(
-                fargs,
-                p,
-                cbmc_ret_ty,
-                n
-            ));
+        if let Some(stripped) = intrinsic.strip_prefix("simd_shuffle") {
+            let n: u64 = if stripped.is_empty() {
+                // Make sure that this is an array, since only the
+                // length-suffixed version of `simd_shuffle` (e.g.,
+                // `simd_shuffle4`) is type-checked
+                match farg_types[2].kind() {
```

Comment:
please assert that fargs.len() == 3

Reply:
The check appeared where we codegen, I've moved it so it's done before all of this.